### PR TITLE
Fix blitz CLI not using proper exit codes

### PR DESCRIPTION
**Pull request by markhaehnel**
Closes: #1303

### What are the changes and their implications?
blitz/cli now returns exit codes of the spawned child process.
This allows for better use of blitz command in CI/CD environments.

### Checklist

- [X] Tests added for changes
- [ ] PR submitted to [blitzjs.com](https://github.com/blitz-js/blitzjs.com) for any user facing changes

### CODE REVIEW

**Comment by jamiedavenport** [comment]
Thanks for opening this. The fix looks good. Could you add some tests for this so that we can avoid regression in future?

**Comment by markhaehnel** [comment]
> Could you add some tests so that we can avoid regression in future?
Sure, i will try. 🙂

**Comment by markhaehnel** [comment]
I've added a basic test that checks the return type but encountered a problem. I've asked on slack for input on some ideas. (https://blitzjs.slack.com/archives/CUM2C68T1/p1602605269128000)

File: packages/blitz/test/bin.test.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  it("should return exit code 0 if command is successfull", () => {
+    const result = spawn.sync("blitz", ["--version"])
+    expect(result.status).toBe(0)
+  })
+  it.skip("should return exit code 1 if command is unsuccessfull", () => {
+    // TODO: Commands not available dont return exit code 1.
+    // Need do setup a environment to get it failing.
+
+    const result = spawn.sync("blitz", ["notAvailableCommand"])
+    expect(result.status).toBe(1)
```
Comment (jamiedavenport) [inline packages/blitz/test/bin.test.ts:13]: I think something like the change below might work 
```suggestion
    const result = spawn.sync("./bin/blitz", ["build"], {stdio: "inherit"})
    expect(result.status).toBe(1)
```

File: packages/blitz/test/bin.test.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import spawn from "cross-spawn"
+
+describe("Binary blitz", () => {
+  it("should return exit code 0 if command is successfull", () => {
+    const result = spawn.sync("blitz", ["--version"])
```
Comment (jamiedavenport) [inline packages/blitz/test/bin.test.ts:5]: I think this also needs to change to `./bin/blitz`. I think just spawning `blitz` might run the globally installed CLI on your machine.

```suggestion
    const result = spawn.sync("./bin/blitz", ["--version"], {stdio: "inherit"})
```

**Comment by markhaehnel** [comment]
Now the test should work, but it doesnt run in CI because the lerna scopa "blitz" doesnt get tested

**Comment by flybayer** [comment]
@markhaehnel thanks! I fixed the blitz pkg tests not running. Looks like your new test is failing with exit code of `2` instead of `1`

**Comment by markhaehnel** [comment]
My tests are running now, but can't explain to me why the example builds fail. 🤷‍♂️
Looks like a timeout.

**Comment by markhaehnel** [comment]
Should be good now. Have a nice weekend!

**Comment by flybayer** [comment]
Sweet, thank you!!

@all-contributors add @markhaehnel for code

**Pull request merged**